src/org/stringtemplate/v4/STGroup.java

```diff
@@ -80,9 +80,11 @@
 		TEMPLATE_FILE_EXTENSION = ".st";
 	}
 
-    /** When we use key as a value in a dictionary, this is how we signify. */
-    public static final String DICT_KEY = "key";
-    public static final String DEFAULT_KEY = "default";
+    /** When we use key as a value in a dictionary, this is how we signify.
+     *  Magic suffix is used to distinguish "default" string literal and "default" key
+     *  End part is the date of this suffix adding. */
+    public static final String DICT_KEY = "key_st4_magic_suffix_2016/01/10";
```

Comment:
The change to `DICT_KEY` shouldn't be necessary, as the object is [compared by reference](https://github.com/antlr/stringtemplate4/blob/cb00ede2bd9ffbdbead442a7cda8c277ffdec8b9/src/org/stringtemplate/v4/misc/MapModelAdaptor.java#L52), not by value.